File: src/main/java/net/torosamy/safetradeywy/pojo/manager/TradeManager.java
package net.torosamy.safetradeywy.pojo.manager;

import net.torosamy.safetradeywy.pojo.trade.Trade;

import java.util.ArrayList;
import java.util.List;


public class TradeManager {
    private static List<Trade> tradeList = new ArrayList<>();


    public static String showAllTrade() {
        String res = "";
        for (Trade trade : tradeList) {
            res += trade.getSender().getName();
            res += " ";
            res += trade.getReceiver().getName();
            res += "\n";
        }
        return res;
    }

    public static Integer findOnTrade(String name) {
        for (int i = 0; i < tradeList.size(); i++) {
            if (!tradeList.get(i).getConfirmRequest()) continue;
            if (tradeList.get(i).getSender().getName().equalsIgnoreCase(name)) return i;
            if(tradeList.get(i).getReceiver().getName().equalsIgnoreCase(name)) return i;
        }
        return -1;
    }

    public static List<Trade> getTradeList() {
        return tradeList;
    }

    public static Integer findBySender(String name) {
        for (int i = 0; i < tradeList.size(); i++) {
            if (tradeList.get(i).getSender().getName().equalsIgnoreCase(name)) return i;
        }
        return -1;
    }

    public static void removeAllBySender(String name) {
        int index = findBySender(name);
        while (index != -1) {
            tradeList.remove(index);
            index = findBySender(name);
        }
    }

    public static Integer findByReceiver(String name) {
        for (int i = 0; i < tradeList.size(); i++) {
            if (tradeList.get(i).getReceiver().getName().equalsIgnoreCase(name)) return i;
        }
        return -1;
    }


    public static void removeAllByReceiver(String name) {
        int index = findByReceiver(name);
        while (index != -1) {
            tradeList.remove(index);
            index = findByReceiver(name);
        }
    }

    public static Integer findBySenderAndReceiver(String sender, String receiver) {
        for (int i = 0; i < tradeList.size(); i++) {
            if (tradeList.get(i).getReceiver().getName().equalsIgnoreCase(receiver) && tradeList.get(i).getSender().getName().equalsIgnoreCase(sender))
                return i;
        }
        return -1;
    }

    public static Boolean checkTradeRepeat(String name1, String name2) {
        for (int i = 0; i < tradeList.size(); i++) {
            if (tradeList.get(i).getReceiver().getName().equalsIgnoreCase(name1) && tradeList.get(i).getSender().getName().equalsIgnoreCase(name2))
                return true;
            if (tradeList.get(i).getReceiver().getName().equalsIgnoreCase(name2) && tradeList.get(i).getSender().getName().equalsIgnoreCase(name1))
                return true;
        }
        return false;

    }


    public static void removeByIndex(int index) {
        if (index == -1) return;
        tradeList.remove(index);
    }
}

File: src/main/kotlin/net/torosamy/safeTrade/manager/TradeManager.kt
package net.torosamy.safeTrade.manager

import net.torosamy.safeTrade.SafeTrade
import net.torosamy.safeTrade.pojo.Trade
import net.torosamy.safeTrade.scheduler.CheckCancelTask
import net.torosamy.safeTrade.utils.ConfigUtil
import net.torosamy.safeTrade.utils.HoverUtil
import org.bukkit.entity.Player

class TradeManager {
    companion object{
        var tradeList = ArrayList<Trade>()


        fun getAllTradeInfo(): String {
            var res = ""
            for (trade in tradeList) { res += trade.sender.name + " " + trade.receiver.name +"\n" }
            return res
        }


        fun addTrade(sender: Player, receiver:Player): Boolean {
            //前置插件会检测玩家是否存在

            val senderIndex = getTradeIndex(sender.name)
            if(senderIndex != -1) {
                val message = HoverUtil.replacePapi(ConfigUtil.langConfig.sendFailSelf,sender.name,receiver.name)
                sender.sendMessage(message)
                return false
            }

            val receiverIndex = getTradeIndex(receiver.name)
            if (receiverIndex != -1) {
                val message = HoverUtil.replacePapi(ConfigUtil.langConfig.sendFailOther,sender.name,receiver.name)
                sender.sendMessage(message)
                return false
            }
            //如果发送成功 则开始监测 以便于自动删除
            val trade = Trade(sender, receiver)
            tradeList.add(trade)
            CheckCancelTask(ConfigUtil.mainConfig.cancelSecond,trade).runTaskTimer(SafeTrade.plugin,0L,20L)
            return true
        }

        fun removeTrade(index: Int):Boolean {
            if(index == -1) return false
            val trade = tradeList.get(index)
            trade.tradeInventory.isFinished = true
            tradeList.remove(trade)
            return true
        }

        fun removeTrade(name: String):Boolean {
            return removeTrade(getTradeIndex(name))
        }

        fun removeTrade(trade: Trade):Boolean {
            return removeTrade(getTradeIndex(trade.sender.name))
        }

        fun getTradeIndex(name:String):Int {
            for ((index, trade) in tradeList.withIndex()) {
                if(trade.receiver.name == name || trade.sender.name == name) return index
            }
            return -1;
        }



    }
}